### PR TITLE
Fix `_document.tsx` in "with-custom-font" example monorepo

### PR DESCRIPTION
**Pull request by derek-primumco**

### CODE REVIEW

File: example-monorepos/with-custom-font/apps/next/pages/_document.tsx [hunk 1]
```diff
@@ -55,6 +55,8 @@ body {
   -moz-osx-font-smoothing: grayscale;
   -ms-overflow-style: scrollbar;
 }
+
+${customFontCss}
```
Comment (derek-primumco) [inline example-monorepos/with-custom-font/apps/next/pages/_document.tsx:59]: I forgot this the first time, so I didn't actually fix anything in the last PR 🙃

Comment (derek-primumco) [reply example-monorepos/with-custom-font/apps/next/pages/_document.tsx:59]: - PR #373

File: example-monorepos/with-custom-font/apps/next/pages/_document.tsx [hunk 2]
```diff
@@ -70,7 +72,7 @@ export async function getInitialProps({ renderPage }) {
   return { ...page, styles: React.Children.toArray(styles) }
 }
 
-export default class Document extends NextDocument {
+export class Document extends NextDocument {
```
Comment (nandorojo) [inline example-monorepos/with-custom-font/apps/next/pages/_document.tsx:75]: This is wrong, not sure why you removed `default`

Comment (derek-primumco) [reply example-monorepos/with-custom-font/apps/next/pages/_document.tsx:75]: @nandorojo I hear you, but `export default Document` is already listed on line 93:

image
https://github.com/nandorojo/solito/pull/374/files/933d4b181199df3cdffe488a1192c31e52e4342c#

Here's a screenshot from VS Code from this branch showing the error -- a module can't have multiple default exports.
image

Comment (nandorojo) [reply example-monorepos/with-custom-font/apps/next/pages/_document.tsx:75]: oh, we can just remove the export then

**Review by derek-primumco — COMMENTED** [review]
@nandorojo As requested

**Comment by andresgutgon** [comment]
Why not use `next/fonts/local` package for the Next part?
https://nextjs.org/docs/pages/building-your-application/optimizing/fonts

**Comment by nandorojo** [comment]
If you create an example where that works with React Native Web, feel free to share.

**Comment by andresgutgon** [comment]
> If you create an example where that works with React Native Web, feel free to share.

Stupid question. The examples in this repo use NextJS. Do they are also using React Native Web? I'm totally new to native dev

**Comment by nandorojo** [comment]
yeah, on the web they use RNW, which is a web library that matches the API of React Native. It might help to read through the docs of Solito and RNW. I also have talks about all this on YouTube, linked on the Solito docs

**Comment by andresgutgon** [comment]
I see. 
``` javascript
import { Text } from 'react-native'
```
In reality is `react-native-web`. This way in my `packages/ui` lib I can import a `<Text />` as it was only for React Native but under the hood is producing DOM elements for the web. 

This is my understanding. Is that the gist of it?

**Comment by nandorojo** [comment]
that’s correct. we’re applying a webpack alias that turns all RN imports into RNW imports.

**Comment by andresgutgon** [comment]
I already spent my geek time this weekend so I will not do a full example sorry 🙏 . I'll left here the main parts of my experiment with `next/font/local` just in case can help someone:

## In a monorepo example...

```javascript
// client/apps/mobile/app/_layout.tsx (Using expo-router)
import { Stack, SplashScreen } from 'expo-router'
import Provider from '@ui/Provider'
import useFonts from 'app/hooks/useFonts'

export default function AppLayout() {
  const areLoaded = useFonts()

  if (!areLoaded) {
    return <SplashScreen />
  }

  return (
    <Provider>
      <Stack />
    </Provider>
  )
}
```

Pick fonts from `web/public` folder (My NextJS app)
``` javascript
// client/apps/mobile/app/hooks/useFonts.ts
import { useFonts as useExpoFonts } from 'expo-font'

const useFonts = () => {
  const [isLoaded] = useExpoFonts({
    Inter_400: require('../../../web/public/font/Inter/Inter-Regular.otf'),
    Inter_500: require('../../../web/public/font/Inter/Inter-Medium.otf'),
    Inter_700: require('../../../web/public/font/Inter/Inter-Bold.otf'),
    Inter_900: require('../../../web/public/font/Inter/Inter-Black.otf'),
  })

  return isLoaded
}

export default useFonts
```

## Diferent tailwind.config for [web] or [mobile]
Mobile Tailwind config
``` javascript
// client/apps/mobile/tailwind.config.js
// eslint-disable-next-line @typescript-eslint/no-var-requires
const preset = require('tailwind-config/tailwind.config.js')

/** @type {import('tailwindcss').Config} */
module.exports = {
  presets: [preset],
  content: [
    './src/pages/**/*.{ts,tsx}',
    '../../packages/ui/**/*.{js,jsx,ts,tsx}',
  ],
  theme: {
    extend: {
      fontFamily: {
        inter400: ['Inter_400'],
        inter500: ['Inter_500'],
        inter700: ['Inter_700'],
        inter900: ['Inter_900'],
      },
    },
  },
}
```



## Next/font integration
This is the part where I use the standard way of loading fonts now in Next:
``` javascript
// client/apps/web/src/pages/_app.tsx
import 'raf/polyfill'
import 'setimmediate'

import cn from 'classnames'
import localFont from 'next/font/local'

const inter = localFont({
  variable: '--font-inter',
  src: [
    {
      path: '../../public/font/Inter/Inter-Regular.otf',
      weight: '400',
      style: 'normal',
    },
    {
      path: '../../public/font/Inter/Inter-Medium.otf',
      weight: '500',
      style: 'normal',
    },
    {
      path: '../../public/font/Inter/Inter-Bold.otf',
      weight: '700',
      style: 'normal',
    },
    {
      path: '../../public/font/Inter/Inter-Black.otf',
      weight: '900',
      style: 'normal',
    },
  ],
})

import '../global.css'

import Head from 'next/head'
import type { SolitoAppProps } from 'solito'
import Provider from '@ui/Provider'

export default function App({ Component, pageProps }: SolitoAppProps) {
  return (
    <>
      <Head>
        <title>Solito Example App</title>
        <meta
          name='description'
          content='Expo + Next.js with Solito. By Fernando Rojo.'
        />
        <link rel='icon' href='/favicon.ico' />
      </Head>
      <Provider>

        <main className={cn(inter.variable)}>
          <Component {...pageProps} />
        </main>
      </Provider>
    </>
  )
}
```

☝️ The most interesting part here is that is using a generated CSS variable that is used in the next tailwind config
``` javascript
const inter = localFont({
  variable: '--font-inter',
  // ...
})
//...
<main className={cn(inter.variable)}>
  <Component {...pageProps} />
</main>
```

Web Tailwind config:
``` javascript
// client/apps/web/tailwind.config.js

// eslint-disable-next-line @typescript-eslint/no-var-requires
const defaultTheme = require('tailwindcss/defaultTheme')
// eslint-disable-next-line @typescript-eslint/no-var-requires
const preset = require('tailwind-config/tailwind.config.js')

/** @type {import('tailwindcss').Config} */
module.exports = {
  presets: [preset],
  content: [
    './src/app/**/*.{ts,tsx}',
    './src/pages/**/*.{ts,tsx}',
    '../../packages/ui/**/*.{js,jsx,ts,tsx}',
  ],
  theme: {
    extend: {
      fontFamily: {
        // Generated CSS variable in _app.tsx Next part
        sans: ['var(--font-inter)', ...defaultTheme.fontFamily.sans],
      },
    },
  },
}
```

## Finally my web/mobile component using the font

``` javascript
client/packages/ui/ds/atoms/Text/index.tsx
import { ComponentProps, forwardRef } from 'react'
import cn from 'classnames'
import { Text as NativeText } from 'react-native'
import { styled } from 'nativewind'

const FONT = {
  web: {
    normal: 'font-sans',
    medium: 'font-sans font-medium',
    bold: 'font-sans font-bold',
    black: 'font-sans font-black',
  },
  mobile: {
    normal: 'font-inter400',
    medium: 'font-inter500',
    bold: 'font-inter700',
    black: 'font-inter900',
  },
}

const WrappedText = styled(NativeText)

type Props = TextProps & {
  fontWeight?: 'normal' | 'medium' | 'bold' | 'black'
  className?: string
}
export const Text = forwardRef<NativeText, Props>(function Text(
  { children, className = '', fontWeight = 'normal', ...props },
  ref,
) {
  const font = Platform.select({
    web: { className: FONT.web },
    default: { className: FONT.mobile },
  }).className
  return (
    <WrappedText
      ref={ref}
      {...props}
      className={cn(className, {
        [font.normal]: fontWeight === 'normal',
        [font.medium]: fontWeight === 'medium',
        [font.bold]: fontWeight === 'bold',
        [font.black]: fontWeight === 'black',
      })}
    >
      {children}
    </WrappedText>
  )
})
```

**Comment by nandorojo** [comment]
Oh cool. Yeah I could see this working with Tailwind since it uses the CSS lib on Web. But I haven’t seen an example for other libraries, so it doesn’t really fit this example for now.

**Comment by andresgutgon** [comment]
Tailwind doesn't matter no? I mean this code is loading the font in mobile or in web. 

I'm not sure what you mean by "other libraries" but imaging it means something like https://www.dripsy.xyz/ you could use the same CSS variable generated here when defining the theme (Never used dripsy so maybe is wrong)

**Comment by nandorojo** [comment]
yeah true that might be possible actually. I just haven’t spent enough time looking into it

**Pull request merged**